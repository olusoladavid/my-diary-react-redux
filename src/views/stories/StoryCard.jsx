import React from "react";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";
import transformDate from "../../utils/dateTransformer";

const StoryCard = ({ story: { id, title, content, created_on }, link }) => {
  const { day, month, year } = transformDate(created_on);
  return (
    <figure className="story-list__item">
      <div className="item__holder" />
      <div className="item__date">
        <span className="date__span date__day">{day}</span>
        <span className="date__span date__month">{month.substring(0, 3).toUpperCase()}</span>
        <span className="date__span date__year">{year}</span>
      </div>
      <figcaption className="item__caption">
        <h3 className="caption__text caption__title">{title}</h3>
        <p className="caption__text caption__content">{content.substring(0, 100)}</p>
      </figcaption>
      <div className="item__overlay">Read Story</div>
      <Link className="link item__link" to={link || `/story/${id}`} />
    </figure>
  );
};

StoryCard.propTypes = {
  story: PropTypes.object.isRequired,
  link: PropTypes.string
};

export default StoryCard;
