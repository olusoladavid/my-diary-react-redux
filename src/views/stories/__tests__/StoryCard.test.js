import StoryCard from "../StoryCard";

const story = {
  id: 1,
  title: "title",
  content: "content",
  created_on: "2018-11-08T15:14:04.276Z"
};

it("renders story content without crashing", () => {
  const wrapper = shallow(<StoryCard story={story} />);
  expect(wrapper).toMatchSnapshot();
});

it("renders story content without crashing", () => {
  const wrapper = shallow(<StoryCard story={story} link="/link" />);
  expect(wrapper).toMatchSnapshot();
});
