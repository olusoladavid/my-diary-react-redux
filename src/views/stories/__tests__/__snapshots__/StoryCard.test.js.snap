// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders story content without crashing 1`] = `
<figure
  className="story-list__item"
>
  <div
    className="item__holder"
  />
  <div
    className="item__date"
  >
    <span
      className="date__span date__day"
    >
      08
    </span>
    <span
      className="date__span date__month"
    >
      NOV
    </span>
    <span
      className="date__span date__year"
    >
      2018
    </span>
  </div>
  <figcaption
    className="item__caption"
  >
    <h3
      className="caption__text caption__title"
    >
      title
    </h3>
    <p
      className="caption__text caption__content"
    >
      content
    </p>
  </figcaption>
  <div
    className="item__overlay"
  >
    Read Story
  </div>
  <Link
    className="link item__link"
    replace={false}
    to="/story/1"
  />
</figure>
`;

exports[`renders story content without crashing 2`] = `
<figure
  className="story-list__item"
>
  <div
    className="item__holder"
  />
  <div
    className="item__date"
  >
    <span
      className="date__span date__day"
    >
      08
    </span>
    <span
      className="date__span date__month"
    >
      NOV
    </span>
    <span
      className="date__span date__year"
    >
      2018
    </span>
  </div>
  <figcaption
    className="item__caption"
  >
    <h3
      className="caption__text caption__title"
    >
      title
    </h3>
    <p
      className="caption__text caption__content"
    >
      content
    </p>
  </figcaption>
  <div
    className="item__overlay"
  >
    Read Story
  </div>
  <Link
    className="link item__link"
    replace={false}
    to="/link"
  />
</figure>
`;
